src/HealthChecks.Publisher.ApplicationInsights/DependencyInjection/ApplicationInsightsHealthCheckBuilderExtensions.cs

```diff
@@ -15,12 +15,14 @@ public static class ApplicationInsightsHealthCheckBuilderExtensions
         /// indicating the health check status (1 - Healthy, 0 - Unhealthy) and the total time the health check took to execute in milliseconds.
         /// </remarks>
         /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
-        /// <param name="instrumentationKey">Specified Application Insights instrumentation key. Optional. If <c>null</c> TelemetryConfiguration.Active is used.</param>
+        /// <param name="connectionString">Specified Application Insights connection string. Optional. If <c>null</c> <paramref name="instrumentationKey"/> is used.</param>
+        /// <param name="instrumentationKey">Specified Application Insights instrumentation key. Optional. If <c>null</c> <see cref="TelemetryConfiguration"/> is resolved from DI container.</param>
         /// <param name="saveDetailedReport">Specifies if save an Application Insights event for each HealthCheck or just save one event with the global status for all the HealthChecks. Optional</param>
         /// <param name="excludeHealthyReports">Specifies if save an Application Insights event only for reports indicating an unhealthy status</param>
         /// <returns>The specified <paramref name="builder"/>.</returns>
         public static IHealthChecksBuilder AddApplicationInsightsPublisher(
             this IHealthChecksBuilder builder,
+            string? connectionString = default,

```

Comment:
This change is breaking. Both at binary level and source level - you should point compiler what arg to pass - `instrumentationKey: ` or `connectionString:`. For someone it can be a problem though I don't see a way to do better decision.

Reply:
So, can we merge this PR?

Reply:
#1533

Reply:
@skyfrk please pull changes from `master` to verify CI. Also now is a good time to go on with any breaking changes if any. See #1574.

Reply:
I synced my fork :)

Reply:
Thanks. Waiting for CI.